server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -228,7 +230,7 @@ private void doRecovery(final long recoveryId, final StartRecoveryRequest preExi
                 new RecoveryFailedException(recoveryTarget.state(), "failed to prepare shard for recovery", e),
                 true
             );
-        }), recoveryRef::close));
+        }), onCompletion::close));
```

Comment:
Just a question for my learning: does this mean also when the preRecovery call fails, we would disable the recovery monitor? If yes, I guess that is unnecessary but harmless since the peer recover has failed?

Reply:
Closing the releasable means we _re-enable_ the monitor. But yes this doesn't really do anything, the recovery is already dead at this point.